src/bun.js/test/jest.zig

```diff
@@ -2771,7 +2771,94 @@ pub const Expect = struct {
 
         const expected_line = "Expected constructor: <green>{any}<r>\n";
         const received_line = "Received value: <red>{any}<r>\n";
-        const fmt = comptime getSignature("toBeInstanceOf", "", false) ++ "\n\n" ++ expected_line ++ received_line;
+        const fmt = comptime getSignature("toBeInstanceOf", "<green>expected<r>", false) ++ "\n\n" ++ expected_line ++ received_line;
+        if (Output.enable_ansi_colors) {
+            globalObject.throw(Output.prettyFmt(fmt, true), .{ expected_fmt, value_fmt });
+            return .zero;
+        }
+
+        globalObject.throw(Output.prettyFmt(fmt, false), .{ expected_fmt, value_fmt });
+        return .zero;
+    }
+
+    pub fn toMatch(this: *Expect, globalObject: *JSC.JSGlobalObject, callFrame: *JSC.CallFrame) callconv(.C) JSValue {
+        defer this.postMatch(globalObject);
+
+        const thisValue = callFrame.this();
+        const _arguments = callFrame.arguments(1);
+        const arguments: []const JSValue = _arguments.ptr[0.._arguments.len];
+
+        if (arguments.len < 1) {
+            globalObject.throwInvalidArguments("toMatch() requires 1 argument", .{});
+            return .zero;
+        }
+
+        if (this.scope.tests.items.len <= this.test_id) {
+            globalObject.throw("toMatch() must be called in a test", .{});
+            return .zero;
+        }
+
+        active_test_expectation_counter.actual += 1;
+
+        var formatter = JSC.ZigConsoleClient.Formatter{ .globalThis = globalObject, .quote_strings = true };
+
+        const expected_value = arguments[0];
+        if (!expected_value.isString() and !expected_value.isRegExp()) {
+            globalObject.throw("Expected value must be a string or regular expression: {any}", .{expected_value.toFmt(globalObject, &formatter)});
+            return .zero;
+        }
+        expected_value.ensureStillAlive();
+
+        const value = Expect.capturedValueGetCached(thisValue) orelse {
+            globalObject.throw("Internal consistency error: the expect(value) was garbage collected but it should not have been!", .{});
+            return .zero;
+        };
+        value.ensureStillAlive();
+
+        if (!value.isString()) {
+            globalObject.throw("Received value must be a string: {any}", .{value.toFmt(globalObject, &formatter)});
+            return .zero;
+        }
+
+        const not = this.op.contains(.not);
+        var pass: bool = brk: {
+            if (expected_value.isString()) {
+                if (value.getPrototype(globalObject).get(globalObject, "includes")) |includes_fn| {
```

Comment:
To avoid getting the prototype, we can use `.toZigString()` with `value` and `expected_value` and use `strings.contains()` with the slices

Reply:
I think it'd be better to do `.getZigString(globalObject)` and then add a "contains" function on ZIgString that avoids the utf16 conversion step (saves memory)

Another option is to expose a `stringContains` binding